includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -76,6 +79,42 @@ public static function my_account_menu_items( $items ) {
 		return array_slice( $items, 0, 1, true ) + $custom_endpoints + array_slice( $items, 1, null, true );
 	}
 
+	/**
+	 * Handle password reset requestl.
+	 */
+	public static function handle_password_reset_request() {
+		if ( ! is_user_logged_in() ) {
+			return;
+		}
+		$nonce = filter_input( INPUT_GET, self::RESET_PASSWORD_URL_PARAM, FILTER_SANITIZE_STRING );
+
+		if ( $nonce ) {
+			$is_error = false;
+			if ( wp_verify_nonce( $nonce, self::RESET_PASSWORD_URL_PARAM ) ) {
+				$result  = retrieve_password( wp_get_current_user()->user_login );
```

Comment:
Ugh, yeah. I don't think we can ever rely on `user_login` being a valid email, it will be escaped. Updated in 11aaeeffe49f27b985b29534e29f6461eedfb209 to use `user_email`.